lib/src/realm_object.dart

```diff
@@ -25,133 +25,156 @@
 
 /// A object in a realm. 
 /// 
 /// RealmObjects are generated from Realm data model classes
 /// A data model class `_MyClass` will have a RealmObject with name `MyClass` generated 
 /// which should be used insead of directly instantiating and working with RealmObject instances
-class RealmObject /*extends DynamicObject*/ {
-  Map<String, Object>? _unmanagedProperties;
-
-  /**
-   *  Default constructor. Enables the subclass to different constructors and work with RealmObject unmanaged instances
-   */
-  RealmObject() {
-    _unmanagedProperties = new Map<String, Object>();
+class RealmObject {
+  String _value = "";
+  void setString(String name, String value) {
+    _value = value;
   }
 
-  /**
-   *  Creates managed RealmObject. Called from generated code
-   */
-  RealmObject.constructor() {}
-
-  Object get _realm native "RealmObject_get__realm";
-
-  Object _native_get(String name) native "RealmObject_get_property";
-  void _native_set(String name, Object value) native "RealmObject_set_property";
-
-  Object operator [](String name) {
-    if (_unmanagedProperties != null) {
-      return _unmanagedProperties![name]!;
-    }
-
-    Object result = _native_get(name);
-    if (result is RealmList) {
-      throw new Exception("Invalid RealmObject. RealmLists should be retrieved using super_get method");
-    }
-
-    return result;
+  String getString(String name) {
+    return _value;
   }
 
-  void operator []=(String name, Object value) {
-    if (_unmanagedProperties != null) {
-      _unmanagedProperties![name] = value;
-      return;
-    }
+  // SchemaObject get schema => SchemaObject("");
+}
 
-    _native_set(name, value);
-  }
+/// An exception being thrown when a Realm operation or Realm object access fails
+class RealmException implements Exception  {
+  final String message;
+
+  RealmException(this.message);
 
-  static dynamic getSchema(String typeName, Iterable<SchemaProperty> properties) {
-    if (properties.length == 0) {
-        throw new Exception("Class ${typeName} should have at least one field with RealmProperty annotation");
-    }
-
-    dynamic schema = DynamicObject();
-    schema.name = typeName;
-    schema.properties = new DynamicObject();
-
-    for (var realmProperty in properties) {
-      dynamic propertyValue = DynamicObject();
-      propertyValue.type = realmProperty.type;
-      propertyValue['default'] = realmProperty.defaultValue ?? null;
-      propertyValue.optional = realmProperty.optional ?? null;
-      propertyValue.mapTo = realmProperty.mapTo ?? null;
-      schema.properties[realmProperty.propertyName] = propertyValue;
-      if (realmProperty.primaryKey ?? false) {
-        schema.primaryKey = realmProperty.propertyName;
-      }
-    }
-
-    return schema;
+  String toString() {
+    return "RealmException: $message";
   }
+}
 
-  Object isValid() native "RealmObject_isValid";
+  // Map<String, Object>? _unmanagedProperties;
+
+  // /**
```

Comment:
Should we remove this out-commented code?

Reply:
yes. lets remove that when read object is ready

Reply:
resolving this one. will remember to clear the code in the following PR